pkg/overlay/server_test.go

```diff
@@ -65,3 +67,142 @@ func TestServer(t *testing.T) {
+	time.Sleep(2 * time.Second)
+
+	satellite := planet.Satellites[0]
+
+	for i, tt := range []struct {
+		name                  string
+		newNodeAuditThreshold int64
+		newNodePercentage     float64
+		requestedNodeAmt      int64
+		expectedResultLength  int
+		reputableNodes        int
+	}{
+		{
+			name:                  "case: fewer than required reputable nodes",
+			requestedNodeAmt:      4,
+			reputableNodes:        3,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: more than required reputable nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        3,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: zero reputable nodes found, only new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        0,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:              "case: fewer than required new nodes",
+			requestedNodeAmt:  2,
+			reputableNodes:    3,
+			newNodePercentage: 0.5,
+			// this gives extra reputable instead
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: more than required new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        2,
+			newNodePercentage:     0.5,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			// todo(nat): fix nodes length issue
+			name:                  "case: zero new nodes found, only reputable nodes",
+			requestedNodeAmt:      3,
+			reputableNodes:        3,
+			newNodePercentage:     0.5,
+			expectedResultLength:  4,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: exactly the required amount of new and reputable nodes returned",
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     1,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:              "case: low percentage of new nodes",
+			requestedNodeAmt:  3,
+			reputableNodes:    1,
+			newNodePercentage: 0.01,
+			// todo(nat): expect this result to be 1
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: high percentage of new nodes",
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     3,
+			expectedResultLength:  4,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: 0% new nodes requested",
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     0,
+			expectedResultLength:  1,
+			newNodeAuditThreshold: 1,
+		},
+	} {
+		server := overlay.NewServer(satellite.Log.Named("overlay"), satellite.Overlay,
+			&pb.NodeStats{}, 2, tt.newNodeAuditThreshold, tt.newNodePercentage)
+
+		for i := 0; i <= tt.reputableNodes; i++ {
+			err := satellite.Overlay.Put(ctx, planet.StorageNodes[i].ID(), pb.Node{
+				Reputation: &pb.NodeStats{AuditCount: 1},
```

Comment:
Here, I think you should say `Reputation: &pb.NodeStats{AuditCount: tt.newNodeAuditThreshold}` rather than explicitly putting `1` in